R/oauth-flow-auth-code.R

```diff
@@ -85,14 +87,18 @@ req_oauth_auth_code <- function(req, client,
 #' Section 4.1. This is the most commonly used OAuth flow where the user is
 #' opens a page in their browser, approves the access, and then returns to R.
 #'
-#' `oauth_flow_auth_code()` is a high-level wrapper that should
-#' work with APIs that adhere relatively closely to the spec. The remaining
-#' low-level functions can be used to assemble a custom flow for APIs that are
-#' further from the spec:
+#' `oauth_flow_auth_code()` is a high-level wrapper that should work with APIs
+#' that adhere relatively closely to the spec. When possible, it redirects users
+#' back to a temporary local webserver to capture the authorization code. When
+#' this is not possible -- for example, when running on a hosted platform like
+#' RStudio Server -- it prompts the user to enter the code manually instead.
```

Comment:
But the package author needs to do something else to make this work, right? Maybe we should replicate the gargle approach on an httr2 web page so we can recommend that as the default?

Reply:
We'd still need to get people to register that standard `httr2` web page with their OAuth2 providers. The reason that `gargle` works so well is that it ships with a default client that has https://www.tidyverse.org/google-callback/ preregistered.